internal/cli/helpers.go

```diff
@@ -387,97 +387,102 @@ func getDeployedPackJobs(c *v1.Client, cfg *cache.PackConfig, deploymentName str
 	return packJobs, jobErrs, nil
 }
 
-func newQueryOpts(c *v1.Client) *v1.QueryOpts {
-	return c.QueryOpts()
-}
-
-// func newWriteOpts(c *v1.Client) *v1.WriteOpts {
-// 	return c.WriteOpts()
-// }
-
-// clientOptionCount is used to help size the arrays that back the slices of
-// v1.ClientOption from the clientOptsFromEnvironment and clientOptsFromFlags
-// funcs. If it becomes lower than the truth, the underlying array could be grown
-// but that's not the end of the world.
-const clientOptionCount = 8
-
 // clientOptsFromCLI emits a slice of v1.ClientOptions based on the environment
 // and flag set passed to the command.
-func clientOptsFromCLI(c *baseCommand) []v1.ClientOption {
+func clientOptsFromCLI(c *baseCommand) *api.Config {
 	// This implementation leverages the fact that flags always take precedence
 	// over environment variables to naively append the flags to the env
 	// settings.
+	conf := api.DefaultConfig()
+
+	clientOptsFromEnvironment(conf)
+	clientOptsFromFlags(c, conf)
+	return conf
+}
 
-	// This will not handle the case where client_key is set in an env var and
-	// client_cert is set via flag, but that is currently the only case where
-	// two settings have a strong affinity.
-	opts := make([]v1.ClientOption, 0, 2*clientOptionCount)
-	opts = append(opts, clientOptsFromEnvironment()...)
-	opts = append(opts, clientOptsFromFlags(c)...)
-	return opts
+// handlBasicAuth checks whether the NOMAD_ADDR string is in the user:pass@addr
+// format and if it is, it returns user, password and address. It returns "", "",
+// address otherwise.
+func handleBasicAuth(s string) (string, string, string) {
+	before, after, found := strings.Cut(s, "@")
+	if found {
+		user, pass, found := strings.Cut(before, ":")
+		if found {
+			return user, pass, after
+		}
+	}
+	return "", "", before
```

Comment:
I was thinking surely there must be a standard library function for this, but I don't see one :thinking: 

The closes is https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/net/http/request.go;l=949 which is about extracting from the Authorization header which isn't the same.

Reply:
yup, went through the same reasoning. Sadly it is what it is.